### PR TITLE
[Doc] Add monitoring guidelines for function and connector

### PR DESCRIPTION
**Pull request by Anonymitaet**
Fix https://github.com/apache/pulsar/issues/9932

### CODE REVIEW

File: site2/docs/deploy-monitoring.md [hunk 1]
```diff
@@ -69,6 +69,26 @@ brk_ml_cursor_persistZookeeperErrors(namespace="", ledger_name="", cursor_name:"
...
+pulsar-admin functions-worker monitoring-metrics
+```
+
+You can collect functions and connectors metrics from `functions-worker` and export the metrics in JSON formats.
+
+```
+pulsar-admin functions-worker function-stats
+```
+
+The aggregated functions and connectors metrics can be exposed in Prometheus formats as below. You can get `FUNCTIONS_WORKER_ADDRESS` and `WORKER_PORT` from the `functions_worker.yml` file.
```
Comment (Anonymitaet) [inline site2/docs/deploy-monitoring.md:86]: @freeznet I add `FUNCTIONS_WORKER_ADDRESS` here, is it correct?

**Comment by Anonymitaet** [comment]
@freeznet @Huanli-Meng @Jennifer88huang could you please help review? Thanks

File: site2/docs/deploy-monitoring.md (hunk 1 above)
Comment (freeznet) [reply site2/docs/deploy-monitoring.md:86]: `FUNCTIONS_WORKER_ADDRESS` is not in `functions_worker.yml`, it is related how user config their functions worker, related to [http://pulsar.apache.org/docs/en/next/functions-worker/](http://pulsar.apache.org/docs/en/next/functions-worker/).

Comment (Anonymitaet) [reply site2/docs/deploy-monitoring.md:86]: @freeznet thanks for the explanations. Should we need to tell readers how to get `FUNCTIONS_WORKER_ADDRESS` here?

File: site2/docs/reference-metrics.md [hunk 2]
```diff
@@ -416,6 +416,50 @@ All the Pulsar Functions metrics are labelled with the following labels:
...
+- *cluster*: `cluster=${pulsar_cluster}`. `${pulsar_cluster}` is the cluster name that you have configured in the `broker.conf` file.
+- *namespace*: `namespace=${pulsar_namespace}`. `${pulsar_namespace}` is the namespace name.
+
+Connector metrics contain **source** metrics and **sink** metrics.
+
+- **Source** metrics
+
+  | Name | Type | Description |
+  |---|---|---|
+  pulsar_source_written_total|Counter|Total number of records written to a Pulsar topic.
```
Comment (Huanli-Meng) [inline site2/docs/reference-metrics.md:434]: I think it's better to use "the total number of...", same comments for the following similar descriptions.

File: site2/docs/reference-metrics.md [hunk 3]
```diff
@@ -416,6 +416,50 @@ All the Pulsar Functions metrics are labelled with the following labels:
...
+  pulsar_source_received_total|Counter|Total number of records received from source.
+  pulsar_source_received_total_1min|Counter|Total number of records received from source in the last 1 minute.
+  pulsar_source_last_invocation|Gauge|The timestamp of the last invocation of the source.
+  pulsar_source_source_exception|Gauge|Exception from a source.
+  pulsar_source_source_exceptions_total|Counter|Total number of source exceptions.
+  pulsar_source_source_exceptions_total_1min |Counter|Total number of source exceptions in the last 1 minute.
+  pulsar_source_system_exception|Gauge|Exception from system code.
+  pulsar_source_system_exceptions_total|Counter|Total number of system exceptions.
+  pulsar_source_system_exceptions_total_1min|Counter|Total number of system exceptions in the last 1 minute.
+  pulsar_source_user_metric_ | Summary|User defined metric.
```
Comment (Huanli-Meng) [inline site2/docs/reference-metrics.md:445]:
```suggestion
  pulsar_source_user_metric_ | Summary|User-defined metrics.
```

**Review by Huanli-Meng — COMMENTED** [review]
Overall looks good to me. Only leave a minor comments.

**Comment by Anonymitaet** [comment]
@Huanli-Meng thanks for your comments, I've incorporated them, PTAL.

File: site2/docs/deploy-monitoring.md (hunk 1 above)
Comment (Anonymitaet) [reply site2/docs/deploy-monitoring.md:86]: @freeznet any feedback?

**Review by Huanli-Meng — APPROVED** [review]
LGTM

File: site2/docs/deploy-monitoring.md (hunk 1 above)
Comment (freeznet) [reply site2/docs/deploy-monitoring.md:86]: I suggest to add a link to functions-worker doc, should enough to let reader understand.

Comment (Anonymitaet) [reply site2/docs/deploy-monitoring.md]: @freeznet OK, I've added.

**Pull request merged**